src/Transformers/BooleanConverter.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Rubix\ML\Transformers;
+
+use Rubix\ML\DataType;
+use Rubix\ML\Exceptions\InvalidArgumentException;
+
+/**
+ * Boolean Converter
+ *
+ * Convert boolean true/false values to continuous or categorical values.
+ *
+ * @category    Machine Learning
+ * @package     Rubix/ML
+ * @author      Zach Vander Velden
+ */
+class BooleanConverter implements Transformer
+{
+    /**
+     * The value used to replace boolean value `true` with. Use a string value to represent the boolean as a categorical
+     * value. Use a float or int to represent the boolean as a continuous value.
+     *
+     * @var string|float|int
+     */
+    protected $trueValue;
+
+    /**
+     * The value used to replace boolean value `false` with. Use a string value to represent the boolean as a categorical
+     * value. Use a float or int to represent the boolean as a continuous value.
+     *
+     * @var string|float|int
+     */
+    protected $falseValue;
+
+    /**
+     * BooleanConverter constructor.
+     * @param mixed $trueValue
+     * @param mixed $falseValue
+     */
+    public function __construct($trueValue = 'true', $falseValue = 'false')
+    {
+        if (!is_float($trueValue) && !is_int($trueValue) && !is_string($trueValue)) {
+            throw new InvalidArgumentException('TrueValue must be'
+                . ' a string, float, or int.');
+        }
+
+        if (!is_float($falseValue) && !is_int($falseValue) && !is_string($falseValue)) {
+            throw new InvalidArgumentException('FalseValue must be'
+                . ' a string, float, or int.');
+        }
+
+        $this->trueValue = $trueValue;
+        $this->falseValue = $falseValue;
+    }
+
+    /**
+     * Return the data types that this transformer is compatible with.
+     *
+     * @internal
+     *
+     * @return list<\Rubix\ML\DataType>
+     */
+    public function compatibility() : array
+    {
+        return DataType::all();
+    }
+
+    /**
+     * Transform the dataset in place.
+     *
+     * @param array[] $samples
+     */
+    public function transform(array &$samples) : void
+    {
+        foreach ($samples as &$sample) {
+            foreach ($sample as &$value) {
+                if (is_bool($value)) {
```

Comment:
One thing we can do to optimize this is to determine the columns that have type "Other" during a fitting process and then only iterate over those columns. Is that something you considered in your approach? If not, perhaps we could try it out to see if it will yield a justifiable performance boost.

I can help you create a benchmark for this Transformer so you we can measure the difference in approaches.